tests/integration_tests/test_model_training_options.py

```diff
@@ -139,6 +139,46 @@ def test_model_progress_save(skip_save_progress, skip_save_model, tmp_path):
         assert len(training_checkpoints) > 0
 
 
+@pytest.mark.parametrize("save_ludwig_config_with_model_weights", [False, True])
+def test_ludwig_config_save(save_ludwig_config_with_model_weights, tmp_path):
+    input_features, output_features = synthetic_test_data.get_feature_configs()
+
+    config = {
+        "input_features": input_features,
+        "output_features": output_features,
+        "combiner": {"type": "concat"},
+        TRAINER: {"epochs": 5, BATCH_SIZE: 128},
+    }
+
+    # create sub-directory to store results
+    results_dir = tmp_path / "results"
+    results_dir.mkdir()
+
+    # run experiment
+    generated_data = synthetic_test_data.get_generated_data()
+    _, _, _, _, output_dir = experiment_cli(
+        training_set=generated_data.train_df,
+        validation_set=generated_data.validation_df,
+        test_set=generated_data.test_df,
+        output_directory=str(results_dir),
+        config=config,
+        skip_save_processed_input=True,
+        skip_save_progress=False,
+        skip_save_unprocessed_output=True,
+        skip_save_model=False,
+        skip_save_log=True,
+        save_ludwig_config_with_model_weights=save_ludwig_config_with_model_weights,
+    )
+
+    # ========== Check for required ludwig-config file =============
+    model_weights_dir = os.path.join(output_dir, "model", "model_weights")
+    files = [f for f in os.listdir(model_weights_dir) if re.match(r"ludwig_config", f)]
```

Comment:
@sanjaydasgupta This is the key place where we would want to check that the name of the configuration file is exactly what we expect it to be -- like a constant string comparison; could you please see if this is feasible?  Thanks!